### PR TITLE
CP-18255: Implement pool_update.precheck function.

### PR DESCRIPTION
**Pull request by huizh**
1. This is an extension written with Python.
2. It will install control package to run precheck.

Signed-off-by: Hui Zhang hui.zhang@citrix.com

### CODE REVIEW

File: scripts/extensions/pool_update.precheck [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+    session = None
+    try:
+        session = XenAPI.xapi_local()
+        session.xenapi.login_with_password('root', '', '', 'Pool_update')
+
+        update = params[1]
+        host = params[2]
+        # attach and get the yum URL
+        yum_url = session.xenapi.pool_update.attach(update, host)
```
Comment (jonludlam) [inline scripts/extensions/pool_update.precheck]: This will work only when the VDI can be attached on this host.

Comment (huizh) [reply scripts/extensions/pool_update.precheck]: The yum_url returned by pool_update.attach is either local yum (VDI attached on this host) or remote yum (VDI not attached on this host). "parse_control_package" will handle either return. Does it make sense?

Thanks.

**Comment by huizh** [comment]
Hi @jonludlam, the attach/detach APIs have been updated according to the latest design. Correspondingly precheck and apply use the latest APIs to implement the functionalities as well.

Could you please review the code?
Thanks.

**Comment by jonludlam** [comment]
I think this looks much nicer now!

**Comment by huizh** [comment]
Thank you @jonludlam.

**Pull request merged**